File: Solutions/910. Smallest Range II/910. Smallest Range II.cpp
class Solution {
public:
    int smallestRangeII(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int res=nums[nums.size()-1]-nums[0],right=nums[nums.size()-1]-k,left=nums[0]+k;
        for(int i=0;i<nums.size()-1;++i){
            int m_ax=max(right,nums[i]+k);
            int m_in=min(left,nums[i+1]-k);
            res=min(res,m_ax-m_in);
        }
        return res;
    }
};

File: Solutions/910. Smallest Range II/910. Smallest Range II.py
class Solution:
    def smallestRangeII(self, nums: List[int], k: int) -> int:
        nums.sort()
        res,right,left=nums[len(nums)-1]-nums[0],nums[len(nums)-1]-k,nums[0]+k,
        for i in range(0,len(nums));
            m_ax=max(right,nums[i]+k)
            m_in=min(left,nums[i+1]-k)
            res=min(res,m_ax-m_in)
        return res